### ISSUE TITLE
Remove the buttonSpacer used to add margin between components

### ISSUE THREAD

**Issue by Sboonny** [freeCodeCamp/freeCodeCamp#48759]
**Describe the bug**
We have a component called buttonSpacer and its propose is to add margin between components. We can remove the need for it and use CSS to add the margin.

https://github.com/Sboonny/freeCodeCamp/blob/1f0070630a101d87a6f3b483b8cdb3173dbb155d/client/src/components/helpers/button-spacer.tsx#L1-L10?plain=1

Usually, I try to think of a reason for it, but the only thing that I can come up with is, this is a relic of the past, and we don't need it anymore.

If there are special cases in which `margin` needs to be added, CSS selectors can handle that. And if it's there a need for consistency between spacing, adding a gap to its parent component should do the trick, as well.

---

Here is how it looks without the buttonSpacer, here is the page https://www.freecodecamp.org/settings

With the buttonSpacer

![image](https://user-images.githubusercontent.com/88248797/207670150-c6af442e-537b-40c3-8b9a-719f065ee733.png)

Without the buttonSpacer

![image](https://user-images.githubusercontent.com/88248797/207669672-52fe7087-1389-461a-bde6-130a21f6b690.png)

**Comment by joeshajan** [comment]
If it needs to be fixed can I take this issue? Can someone assign this to me?

**Comment by Sboonny** [comment]
This isn't a simple issue because it will require a lot of refactorings, and the button may have a usage that I don't know about. I will wait for other maintainers or prominent contributors to tune in and say what they think about this.

**Comment by dinesh-14** [comment]
@Sboonny Can we get to do a check by removing the buttonSpacer and adding space using css. If refactorings are needed we can track and fix them ?

**Comment by Sboonny** [comment]
I have did the check, to be honest I don't want to open it in holiday season, while other maintainers taken time off for holiday @dinesh-14

**Comment by dinesh-14** [comment]
Let me know once you open.

**Comment by ahmadabdolsaheb** [comment]
Could we use `spacer` helper? perhaps modify it so it gives the desired seperation for components that are using   `buttonSpacer`?

**Comment by Sboonny** [comment]
I will be honest, I am aiming of deleting it too, in future PRs.

https://github.com/Sboonny/freeCodeCamp/blob/main/client/src/components/helpers/spacer.tsx

---

It's understandable when we had it in the past, because we couldn't make sure that the spacing is consistent. Now, we can achieve the same result with applying `gap` property to the parent, and have an easier to read layout than the current one.

**Comment by ahmadabdolsaheb** [comment]
I also suggest looking into handling `spacing in component libraries` in general just to make sure we are heading in the right direction spacing wise.

**Comment by Sboonny** [comment]
Yes, reduce the usage of basic CSS, because it does cause the code base to be bloated, but we are currently a kinda bloated codebase, and I am worried that it will get worse. 

I have no issue with making `Spacer` smarter, but we can go thinner down the line.

---

I have pushed this outside the scope of this PR, but let me say this so take sure we are on the same page.

There are components that have `div` elements wrapping them, because they don't accept className, we can make them smarter and let them accept `className` which will clear some HTML.

We can change the components that use `Spacer` and `buttonSpacer` to have a parent with flex or grid layout, then add `gap`, and when we have more of them, we turn these into a component, that can be used. 

So we can use new solution and reduce old ones, and while on the way remove the unused css properties and duplicated components.

**Comment by shohan2001** [comment]
Give me this issue

**Comment by Sboonny** [comment]
To solve this issue, follow Ahmad suggestion and make `Spacer` smarter (can have 5px as padding top).

---

If you would like to fix this issue, please make sure you read [our guidelines for contributing](https://contribute.freecodecamp.org/#/), we prioritize contributors following the instructions in our guides. Join us in [our chat room](https://discord.gg/PRyKn3Vbay) or [the forum](https://forum.freecodecamp.org/c/contributors/3) if you need help contributing, our moderators will guide you through this.

@dinesh-14, the issue is open.

@shohan2001, feel free to open a PR, after you have read the guideline.

**Comment by BalanaguYashwanth** [comment]
Got it
